component-library/component-lib/src/shared/directives/dom-change.directive.ts

```diff
@@ -0,0 +1,33 @@
+import { Directive, ElementRef, EventEmitter, OnDestroy, Output } from '@angular/core';
+
+/**
+ * @see https://stackblitz.com/edit/angular-mutationobserver-example?file=app%2Fapp.module.ts,app%2Fdom-change.directive.ts,app%2Fapp.component.ts
+ */
+@Directive({
+  selector: '[domChange]'
+})
+export class DomChangeDirective implements OnDestroy {
+  private changes: MutationObserver;
+
+  @Output()
+  public domChange = new EventEmitter();
+
+  constructor(private elementRef: ElementRef) {
+    const element = this.elementRef.nativeElement;
+
+    this.changes = new MutationObserver((mutations: MutationRecord[]) => {
```

Comment:
I didn't realize this was a thing - it may be possible to pass back an observable that removes the need for a listener. Interesting approach!